packages/snap-preact-components/src/components/Molecules/Radio/Radio.tsx

```diff
@@ -0,0 +1,188 @@
+/** @jsx jsx */
+import { h } from 'preact';
+import { useState, StateUpdater } from 'preact/hooks';
+
+import { jsx, css } from '@emotion/react';
+import classnames from 'classnames';
+import { observer } from 'mobx-react-lite';
+
+import { ComponentProps, StylingCSS } from '../../../types';
+import { defined } from '../../../utilities';
+import { Theme, useTheme, CacheProvider } from '../../../providers';
+import { Icon, IconProps } from '../../Atoms/Icon';
+import { useA11y } from '../../../hooks/useA11y';
+import { mergeProps } from '../../../utilities';
+
+const CSS = {
+	radio: ({ size, color, theme }: Partial<RadioProps>) =>
+		css({
+			display: 'inline-flex',
+			alignItems: 'center',
+			justifyContent: 'center',
+			height: size,
+			width: size,
+			cursor: 'pointer',
+
+			'&.ss__radio--disabled': {
+				opacity: 0.5,
+			},
+			'& .ss__radio__empty': {
+				display: 'inline-block',
+				width: `calc(${size} - 30%)`,
+				height: `calc(${size} - 30%)`,
+			},
+
+			'& .ss__radio__icon--active circle:first-of-type': {
+				stroke: `${color || theme?.colors?.primary || '#333'}`,
+			},
+			'& .ss__radio__icon--active circle:last-of-type': {
+				fill: `${color || theme?.colors?.primary || '#333'}`,
+			},
+			'& svg.ss__radio__icon--inactive circle': {
+				stroke: `${color || theme?.colors?.primary || '#333'}  !important`,
+			},
+		}),
+	native: ({}) => css({}),
+};
+
+export const Radio = observer((properties: RadioProps): JSX.Element => {
+	const globalTheme: Theme = useTheme();
+	const defaultProps: Partial<RadioProps> = {
+		size: '20px',
+		startChecked: false,
+		disableA11y: false,
+	};
+
+	const props = mergeProps('radio', globalTheme, defaultProps, properties);
+
+	const {
+		checked,
+		color,
+		disabled,
+		checkedIcon,
+		unCheckedIcon,
+		onClick,
+		size,
+		startChecked,
+		native,
+		disableA11y,
+		disableStyles,
+		className,
+		style,
+		styleScript,
+		theme,
+	} = props;
+
+	const subProps: RadioSubProps = {
+		activeIcon: {
+			// default props
+			className: 'ss__radio__icon--active',
+			icon: checkedIcon || 'bullet',
+			// global theme
+			...globalTheme?.components?.icon,
+			// inherited props
+			...defined({
+				color: color || theme?.colors?.primary,
+				disableStyles,
+				size: size && `calc(${size} - 30%)`,
+			}),
+			// component theme overrides
+			theme: props.theme,
+		},
+		inactiveIcon: {
+			// default props
+			className: 'ss__radio__icon--inactive',
+			icon: unCheckedIcon || 'bullet-o',
+			// global theme
+			...globalTheme?.components?.icon,
+			// inherited props
+			...defined({
+				color: color || theme?.colors?.primary,
```

Comment:
I think the `Icon` component already does this fallback, so this may be unnecessary.